RTL fixes

yosilevy opened the pull request:
Kudos on this great card. Started using it today. Like much of HA it lacks RTL support which I have been slowly fixing.

Changes include:
- Using ha-slider which I created quite a few HA versions ago (in the main repo) to fix slider pin problems.
- Settings directions where needed to keep controls functioning (otherwise slider works the wrong way and pin text is reversed).
- Style updates so components looks right.

Tested in both LTR and RTL to make sure nothing broken.

Before fix (see highlights for problem):
![image](https://user-images.githubusercontent.com/37745463/55458130-c19a3200-55f4-11e9-94e4-f1b079f630b1.png)

After fix:
<img width="336" alt="Capture" src="https://user-images.githubusercontent.com/37745463/55458043-7c760000-55f4-11e9-8582-3b93b6ee777f.PNG">

kalkih commented:
Oh, I actually wasn't aware that the whole UI is flipped when using an RTL locale.

Not sure if I wanna switch from paper-slider to ha-slider as it will break for people running older versions of HA where ha-slider isn't available.

The control strip (volume slider & media buttons) looks weird in your RTL screenshots, it should probably be left aligned.

yosilevy commented:
Fair enough. Reverted back to paper-slider + added patch to the style (they have a known bug and this is how we fixed it in HA main repo).
Went through all styles and added [rtl] styles to match.

Looks great now.

![image](https://user-images.githubusercontent.com/37745463/55529519-793d4b80-56aa-11e9-9eab-010ec36736d8.png)

kalkih commented:
Awesome, that looks much better indeed!

kalkih commented on src/style.js:
```diff
@@ -465,7 +490,7 @@ const style = css`
   .speaker-select > paper-checkbox > span {
     font-weight: 600;
   }
-  paper-slider {
```
Need to revert all these selectors back to paper-slider

yosilevy replied:
Please look again. All references have already been reverted. I think you looked at the wrong file.

kalkih replied:
The HTML yes, but not the CSS selectors.

yosilevy replied:
Where do you see in the CSS ha-paper-slider? They are removed.

kalkih replied:
ha-paper-slider? You changed some CSS selectors to ha-slider, you have to revert those back to paper-slider, I'll make comments on them for you.

yosilevy replied:
Yes. Sorry. Got confused with something else I was doing in HA main repo. Sorry.

kalkih replied:
No worries, are there both a ha-paper-slider and a ha-slider in core?

yosilevy replied:
Yes unfortunately

kalkih commented on src/main.js:
```diff
@@ -173,6 +175,7 @@ class MiniMediaPlayer extends LitElement {
         artwork=${config.artwork} ?has-artwork=${artwork} state=${this.player.state}
         ?flow=${config.flow} ?collapse=${config.collapse}
         content=${this.player.content}
+        ?rtl=${this._computeRTL(this.hass)}
```
What do you think about making this check once inside `set hass()` instead and save the result as a boolean in a member variable (this.rtl) instead? Then we could resure that variable throughout the rendering logic. Like below.

https://github.com/kalkih/mini-media-player/blob/bc848c6e59cc100e8f8b661b036ae552df148b8a/src/main.js#L57-L61

kalkih commented on src/main.js:
```diff
@@ -488,6 +503,7 @@ class MiniMediaPlayer extends LitElement {
           @change=${e => this.player.setVolume(e)}
           @click=${e => e.stopPropagation()}
           min='0' max=${this.config.max_volume} value=${this.player.vol * 100}
+          dir=${this._computeRTLDirection(this.hass)}
```
If we take the member variable approach, we could do 
```js
?rtl=${this.rtl}
```
here. Would have to change the slider patch slightly for the selector to work.

kalkih commented on src/main.js:
```diff
@@ -148,6 +148,8 @@ class MiniMediaPlayer extends LitElement {
     setTimeout(() => this.initial = false, 250);
     this.edit = this.config.speaker_group.expanded || false;
+
+    this.patchSliderForRTL();
```
We only really need to patch this if rtl is `true`
```js
if (this.rtl) this.patchSliderForRTL();
```

kalkih commented on src/style.js:
```diff
@@ -410,7 +431,11 @@ const style = css`
+    margin-right: auto;
+    margin-left: 0;
+  }
+  .control-row--top ha-slider {
```
Should be paper-slider

kalkih commented on src/style.js:
```diff
@@ -465,7 +490,7 @@ const style = css`
     font-weight: 600;
   }
-  paper-slider {
+  ha-slider {
```
Should be paper-slider

kalkih commented on src/style.js:
```diff
@@ -570,26 +595,38 @@ const style = css`
     display: none;
   }
-  ha-card[flow] paper-slider,
+  ha-card[flow] ha-slider,
```
Should be paper-slider

yosilevy commented:
All fixed as you suggested. Works fine.

yosilevy commented:
Was almost working fine. It wouldn't patch the slider if it was rendered after the card was already rendered. Added it to update method + marked as patched or not using classes.

yosilevy commented:
Anything else pending?

kalkih commented on src/main.js:
```diff
@@ -244,6 +255,18 @@ class MiniMediaPlayer extends LitElement {
+  }
+
+  _computeRTLDirection(hass) {
+    return this._computeRTL(hass) ? "rtl" : "ltr";
```
No need to compute it again, we got it in `this.rtl`

kalkih commented:
My last comment + fixing the linter errors, rest looks good!

yosilevy commented:
Do you know why code formatting doesn't work in this project? Does it work for you? That's why I didn't notice those issues.

kalkih commented:
Not sure, should work fine as you can see in the Travis build. Run `npm run lint` and see what error you get.

yosilevy commented:
Please let me know once you merge and release a version. Some RTL consumers are waiting for it... :-)

kalkih commented:
Merged, will be included in the next release.

Great job, thanks for the PR! 🎉
